### PR TITLE
Giving super access to all organizations (PROJQUAY-4456)

### PR DESCRIPTION
**Pull request by bcaton85**
Populates organization list with all organizations when the superuser has full access. Requires API changes from this related PR: https://github.com/kleesc/quay/pull/2.

### CODE REVIEW

**Comment by HammerMeetNail** [comment]
@bcaton85 I'm seeing the api call fire twice. This happens in other parts of the code and we have plans to clean it up later. Think this can be addressed easily in the PR or should it be part of the larger cleanup?
image

File: src/routes/OrganizationsList/OrganizationsList.tsx [hunk 1]
```diff
@@ -26,6 +26,8 @@ import {Suspense, useEffect, useState} from 'react';
 import {
   bulkDeleteOrganizations,
   fetchAllOrgs,
+  fetchOrgsAsSuperUser,
+  SuperUserOrganizations,
```
Comment (syed) [inline src/routes/OrganizationsList/OrganizationsList.tsx:30]: can we normalize the response from the API in the Resource file instead? We should only be exposing `IOgranization` ideally

Comment (bcaton85) [reply src/routes/OrganizationsList/OrganizationsList.tsx]: Sounds good, edited to return an organization list: `IOgranization[]`

**Comment by bcaton85** [comment]
> I'm seeing the api call fire twice. This happens in other parts of the code and we have plans to clean it up later. Think this can be addressed easily in the PR or should it be part of the larger cleanup?
@HammerMeetNail Right now the page load for organizations takes some time so I think we're going to have to refactor anyway. There's a couple of opportunities for optimization that it would probably go better with.

File: src/resources/OrganizationResource.ts [hunk 2]
```diff
@@ -27,6 +26,19 @@ export async function fetchOrg(orgname: string) {
...
+  organizations: IOrganization[];
+}
+
+export async function fetchOrgsAsSuperUser() {
+  const superUserOrgsUrl = `/api/v1/superuser/organizations/`;
+  const response: AxiosResponse<SuperUserOrganizations> = await axios.get(
+    superUserOrgsUrl,
+  );
+  assertHttpCode(response.status, 200);
+  return response.data.organizations;
```
Comment (syed) [inline src/resources/OrganizationResource.ts:39]: null check here
`response.data?.organizations`

File: src/resources/UserResource.ts [hunk 3]
```diff
@@ -41,3 +42,14 @@ export async function fetchUser() {
...
+
+export interface AllUsers {
+  users: IUserResource[];
+}
+
+export async function fetchUsersAsSuperUser() {
+  const superUserOrgsUrl = `/api/v1/superuser/users/`;
+  const response: AxiosResponse<AllUsers> = await axios.get(superUserOrgsUrl);
+  assertHttpCode(response.status, 200);
+  return response.data.users;
```
Comment (syed) [inline src/resources/UserResource.ts:54]: `response.data?.users`

File: src/resources/OrganizationResource.ts (hunk 2 above)
Comment (bcaton85) [reply src/resources/OrganizationResource.ts]: Added

File: src/resources/UserResource.ts (hunk 3 above)
Comment (bcaton85) [reply src/resources/UserResource.ts]: Added

**Review by HammerMeetNail — APPROVED** [review]

**Pull request merged**